pkgs/crane-utils/src/bin/crane-resolve-workspace-inheritance.rs

```diff
@@ -265,6 +471,7 @@ mod tests {
             version = "some version"
 
             [workspace.dependencies]
+            # workspace comments are not copied
```

Comment:
FWIW I'm totally fine to keep all comments in these `Cargo.toml`s (that is to say we don't need to go out of our way to strip them out, i.e. drop the `fmt` module entirely).

Stripping comments is desired when handling building/developing a project itself (to avoid accidental rebuilds), but keeping them in for any git-deps doesn't hurt since the only time they would change is if the dependency's commit is updated in the lock file (which would end up rebuilding anyway)!

Reply:
I updated the comment to clarify: the way we are performing the merge, the "surrounding decoration" is only copied over values that need to be added.   Since this top-level comment is on a key "foo" which is in common with cargo_toml, the keys are merged and the comment is taken from the "cargo_toml" side.

I also removed the `fmt` module in a separate commit, to show how the resulting `smoke_test` truth has to change.  Do you think this is OK as-is (string comparison), or should I try using the `toml` crate as a dev-dependency to compare just TOML content, to restore more sane spacing?

Reply:
Ah thanks for clarifying! Yes I think it's totally fine to keep as is for now (even with the non-standard spacing, thanks for noting that in a comment). We can always go with the `toml`-dev-dependency approach later if this code needs to change!